Lib/test/test_configparser.py

```diff
@@ -760,6 +760,15 @@ def test_read_returns_file_list(self):
         cf = self.newconfig()
         parsed_files = cf.read([], encoding="utf-8")
         self.assertEqual(parsed_files, [])
+        # check when we pass invalid file names:
+        cf = self.newconfig()
+        parsed_files = cf.read([
+            '\x00',                 # NUL bytes filename
+            __file__ + '\x00.ini',  # filename with embedded NUL bytes
+            "\uD834\uDD1E.ini",     # surrogate codes (MUSICAL SYMBOL G CLEF)
+            'a' * 1_000_000         # very long filename
```

Comment:
I was wondering of some of these invalid filenames had already been made into a fixture, so I searched the code and learned that these names were copied from test_linecache. We should avoid copying code as that can lead to divergence and a maintenance burden.

Note also how `test_invalid_names` there uses `subTest` to create separate subtests for each of the inputs. This test should do something similar.

Reply:
Ideally, I would like to, but I wanted to do everything in one PR (I mean, the fixture). Since I didn't know which PR would be accepted first, I preferred doing it separately as well.